File: tests/corpora/reflection/clean_bases/go/calculator.go
// Clean closed-world base - synthetic, no reflection / dynamic dispatch.
// Used as an injection target by pipeline/inject_reflection.py.
// Ground truth (pre-injection): closed-world.
package clean

func add(a, b int) int {
	return a + b
}

func subtract(a, b int) int {
	return a - b
}

func multiply(a, b int) int {
	return a * b
}

func Run(x, y int) int {
	s := add(x, y)
	d := subtract(x, y)
	return multiply(s, d)
}

File: tests/corpora/reflection/clean_bases/php/Calculator.php
<?php
// Clean closed-world base - synthetic, no reflection / dynamic dispatch.
// Used as an injection target by pipeline/inject_reflection.py.
// Ground truth (pre-injection): closed-world.

class Calculator
{
    public function add($a, $b)
    {
        return $a + $b;
    }

    public function subtract($a, $b)
    {
        return $a - $b;
    }

    public function multiply($a, $b)
    {
        return $a * $b;
    }

    public function run($x, $y)
    {
        $s = $this->add($x, $y);
        $d = $this->subtract($x, $y);
        return $this->multiply($s, $d);
    }
}
